Exdi/exdigdbsrvsample/Start-ExdiDebugger.ps1

```diff
@@ -0,0 +1,314 @@
+    }
+
+    New-Item -ItemType Directory $InstallTo -ErrorAction SilentlyContinue | Write-Verbose
+    Copy-Item -Path "$InstallFrom\ExdiGdbSrvSample.dll" -Destination $InstallTo -ErrorAction stop | Write-Verbose
+    Copy-Item -Path "$InstallFrom\exdiConfigData.xml" -Destination $InstallTo -ErrorAction stop | Write-Verbose
+    Copy-Item -Path "$InstallFrom\systemregisters.xml" -Destination $InstallTo -ErrorAction stop | Write-Verbose
+    regsvr32 /s "$InstallTo\ExdiGdbSrvSample.dll"
+
+    if ($(Get-ExdiInstallPath) -eq $null)
+    {
+        throw "Unable to install exdi server"
+    }
+}
+
+Function Edit-ExdiConfigFile
+{
+    [CmdletBinding()]
+    param
+    (
+        [string] $ExdiFilePath,
+        [string] $ExdiTargetType,
+        [PSCustomObject[]] $XmlSettingPathValueList
+    )
+    
+    # Edit exdiConfigData.xml
+    [xml]$exdiConfigXml = Get-Content "$ExdiFilePath"
+
+    # Set current target
+    $exdiConfigXml.ExdiTargets.CurrentTarget = $ExdiTarget
+
+    # set HostNameAndPort
+    $ExdiTargetXmlNode = $exdiConfigXml.SelectSingleNode("//ExdiTargets/ExdiTarget[@Name='$ExdiTarget']/ExdiGdbServerConfigData")
+
+    foreach ($XmlSettingPathValue in $XmlSettingPathValueList)
+    {
+        Write-Verbose "Processing $XmlSettingPathValue"
+        if ($XmlSettingPathValue.Value -eq $null)
+        {
+            continue
+        }
+
+        $PathParts = $XmlSettingPathValue.Path.Split(".")
+        $curNode = $ExdiTargetXmlNode
+        if ($PathParts.Count -gt 1)
+        {
+            foreach ($PathPart in $PathParts[0..($PathParts.Count-2)])
+            {
+                Write-Verbose $PathPart
+                $curNode = $curNode.($PathPart)
+            }
+        }
+        $curNode.($PathParts[-1]) = $XmlSettingPathValue.Value
+    }
+
+    $exdiConfigXml.Save("$ExdiFilePath")
+}
+
+Function Stop-ExdiContainingDllHosts
+{
+    $DllHostPids = Get-Process dllhost | ForEach-Object { $_.Id }
+    foreach ($DllHostPid in $DllHostPids)
+    {
+        $DllHostExdiDlls = Get-Process -Id $DllHostPid -Module | Where-Object { $_.FileName -like "*ExdiGdbSrvSample.dll" }
+        if ($DllHostExdiDlls.Count -ne 0)
+        {
+            Write-Verbose "Killing dllhost.exe with pid $DllHostPid (Contained instance of ExdiGdbSrvSample.dll)"
+            Stop-Process -Id $DllHostPid -Force
+        }
+    }
+}
+
+#endregion
+
+#region Script
+
+Test-ParameterValidation
+
+# look clean up dllhost.exe early since it can hold a lock on files which
+# need to be overwritten
+if (-not $DontTryDllHostCleanup)
+{
+    Stop-ExdiContainingDllHosts
+}
+
+if (-not $(Test-ExdiServerInstalled))
+{
+    if (-not $ExdiDropPath)
+    {
+        throw "ExdiServer is not installed and -ExdiDropPath is not valid"
+    }
+
+    if ($DebuggerPath -like "*DbgX.Shell.exe")
+    {
+        # Cannot install exdi server in appx dir so use localappdata
+        $ExdiInstallDir = "$env:LOCALAPPDATA\DBG\exdi"
```

Comment:
Coinsider using this in all cases. At some point in the near future we will want to switch over to using appx internally as well.

Reply:
Simplified this. The $env:LOCALAPPDATA\Microsoft\WindowsApps dir seems to be writable